### ISSUE TITLE
AstrometricFrame should wrap longitudes to -180 to 180

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#5002]
The AstrometricFrame currently wraps longitudes to the range 0 to 360:

```
<SkyCoord (AstrometricICRS: rotation=0.0 deg, origin=<ICRS Coordinate: (ra, dec) in deg
    (266.40498829, -28.93617776)>): (lon, lat) in deg
    [(0.78239078, -1.37645176), (357.8364555, -4.21280028),
     (2.67949142, -1.79884371), (354.25967185, -1.13503046),
     (2.06089191, -4.54555418), (357.20109258, 0.42632721),
     (1.18017391, -4.32400233), (358.00291391, -1.26503428),
     (3.19026387, -0.39093311), (1.0480904, 2.47843479)]>
```

but I think it should be -180 to 180 by default, so that it really is an offset.

@eteq @alexrudy - what do you think?

**Comment by mhvk** [comment]
Would seem to make more sense to me.

**Comment by mcara** [comment]
I am not sure what ``AstrometricFrame`` is used for and I am no authority in "astrometric frames", however, by looking at Fig. 6-34 in http://adsabs.harvard.edu/abs/2002A%26A...395.1077C I think longitudes should be in range [0,360) unless ``AstrometricFrame`` is intended for mapping Earth: http://futchfactor.com/upload/oz/Map/World/world-map_outline.jpg .

**Comment by mcara** [comment]
In addition, [0,360) is consistent with both ``astropy.wcs`` as well as WCSLIB (used by ``astropy.wcs``)

**Comment by mcara** [comment]
Actually, if used to map Earth, it would be nice to have lon in [0,180] range and add 'W' or 'E' to the number 😄

**Comment by astrofrog** [comment]
Astrometric frames are used to show offset coordinates from a certain position, so you shouldn't think of the coordinates as longitude/latitude, but instead longitude offset and latitude offset.

**Comment by mcara** [comment]
I see... Can it be renamed to ``AstrometricOffsetFrame``?

**Comment by alexrudy** [comment]
This makes sense to me... not sure how to set this up. Does one of the other frames do this?

**Comment by eteq** [comment]
@mcara - I see your point on the overloaded meaning of "Astrometric Frame".  

In some offline discussion, @astrofrog, @lauralwatkins, and a couple others I talked to settled on `SkyOffsetFrame`.  That's reasonably short, but clues the reader into the critical subtlties that 1. it's an offset version of something else and 2. the "offset" is actually in the plane of the sky, not a *3D* offset, and perhaps most critically 3. there are enough subteties that they'd need to look at the documentation and understand what the name actually meant before actually using it.

So I (and the others I mentioned above) am 👍  for changing it to `SkyOffsetFrame`.

I also agree that the [-180,180] change is a good idea.  I think I know how to make this work, so I'll try to give it a shot ASAP.

**Comment by mhvk** [comment]
@eteq - be sure to rebase -- I just merged the clean-ups in #4992.
